python/kiss_icp/kiss_icp.py

```diff
@@ -51,13 +53,14 @@ def register_frame(self, frame, timestamps):
         # Voxelize
         source, frame_downsample = self.voxelize(frame)
 
-        # Get motion prediction and adaptive_threshold
-        sigma = self.get_adaptive_threshold()
+        # Get adaptive_threshold
+        sigma = self.adaptive_threshold.get_threshold()
 
         # Compute initial_guess for ICP
-        prediction = self.get_prediction_model()
-        last_pose = self.poses[-1] if self.poses else np.eye(4)
-        initial_guess = last_pose @ prediction
+        initial_guess = self.current_pose @ self.current_delta
+
+        # Save current pose before the ICP loop to compute the current_delta_ later
+        last_pose = self.current_pose
```

Comment:
This should also be unnecessary. If it's just about avoiding conflicts with current/new naming, we can also rename `self.current_pose` to `self.last_pose` because the current pose is also the last one we computed.

Reply:
Same as https://github.com/PRBonn/kiss-icp/pull/318#discussion_r1560595602